src/App.js

```diff
@@ -84,6 +85,47 @@ const PageNavigation = (params) => {
   }
 };
 
+const AdCluster = (params) => {
+  const isGetAdsRequestPending = params.isGetAdsRequestPending;
+  const isAdDataEmpty = params.isAdDataEmpty;
+  const ads = params.ads;
+  const handleShowNeedLoginModal = params.handleShowNeedLoginModal;
+  const resultsOffset = params.resultsOffset;
+  const getPreviousPageOfAds = params.getPreviousPageOfAds;
+  const getNextPageOfAds = params.getNextPageOfAds;
+
+  if(!isGetAdsRequestPending) {
+    return (
+      <div align="center"><br /><br /><ReactLoading type="spin" color="#000"/></div>
+    );
+  }
+  else {
```

Comment:
nit: since the above if block can only result in a return statement (and thus no other exit from this function/component) this else is not necessary. IMO it's easier to read without the else statement.